api/envoy/api/v2/core/base.proto

```diff
@@ -76,10 +77,47 @@ message Locality {
   string sub_zone = 3;
 }
 
+// BuildVersion combines SemVer version of extension with free-form build information
+// (i.e. 'alpha', 'private-build') as a set of strings.
+message BuildVersion {
+  // SemVer version of extension
+  type.SemanticVersion version = 1;
+
+  // Free-form build information i.e. 'alpha'
+  repeated string labels = 2;
```

Comment:
One final bike shed: should this be key-vale, i.e. a `Struct`? At that point, it can be called "metadata".

Reply:
done